src/components/table/mixins/sort.js

```diff
@@ -0,0 +1,62 @@
+import { EVENT_THEAD_CLICKED } from "../helpers/constants"
+import "./sort.scss"
+
+const SORT_DIRECTION_ASC = "asc"
+const SORT_DIRECTION_DESC = "desc"
+
+export default {
+  data() {
+    return {
+      sortBy: null,
+      sortDir: SORT_DIRECTION_ASC,
+    }
+  },
+  created() {
+    if (this.isSortable) {
+      this.$on(EVENT_THEAD_CLICKED, this.handleSort)
+    }
+  },
+  computed: {
+    sortedData() {
+      const data = [...this.data]
+      return data.sort((a, b) => {
+        const modifier = this.sortDir === SORT_DIRECTION_ASC ? 1 : -1
+        if (a[this.sortBy] < b[this.sortBy]) return -1 * modifier
+        if (a[this.sortBy] > b[this.sortBy]) return 1 * modifier
+        return 0
+      })
+    },
+    isSortable() {
+      return this.fields.some(f => f.sortable)
+    },
+  },
+  methods: {
+    fieldIsSortable(field) {
+      const { sortable } = field
+      return !!sortable
+    },
+    handleSort(field) {
+      if (!this.fieldIsSortable(field)) {
+        return
+      }
+
+      if (field.name === this.sortBy) {
+        this.sortDir =
+          this.sortDir === SORT_DIRECTION_ASC ? SORT_DIRECTION_DESC : SORT_DIRECTION_ASC
+      }
+      this.sortBy = field.name
```

Comment:
Please reset `this.sortDir` to our default value (whatever that is) in the next line. When I change the sort field, I'd expect the sort direction to reset itself. I.e. something like `this.sortDir = SORT_DIRECTION_ASC`.

Reply:
if the second column gets sorted the data from column 1 depends on it and is not sorted anymore. this is why all other columns expect the sorted one gets 

```
aria-sort="none"
```

https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaSort